Model/Queue/Handler/Handler.php

```diff
@@ -115,6 +125,10 @@ public function execute(int $id)
             if (strpos($payment->getMethod(), Method::PAYMENT_TITLE_PREFIX) !== 0) {
                 return;
             }
+            if ($payload['event_type'] == Complete::TYPE) {
```

Comment:
I think it might be good to get this updated now to check if type is Refund Completed, then do this, else if payment completed do the payment logic. And if anything else just return 200 and not process it? As if we add another type in the future, it will still try and do the payment logic, this could future proof it for now.

Reply:
I was thinking it's also worth wrapping the PAYMENT_COMPLETED event type in a similar if / switch case so if we add a new type then it doesn't run the PAYMENT_COMPLETED logic.